sympy/integrals/risch.py

```diff
@@ -237,12 +238,14 @@ def __init__(self, f=None, x=None, handle_first='log', dummy=False, extension=No
         sympows = set()
         logs = set()
         symlogs = set()
+        tans = set()
+        symtans = set()
```

Comment:
These sets could probably be internal to `_rewrite_tans` where functions with rationally related arguments are represented in terms of a single one.